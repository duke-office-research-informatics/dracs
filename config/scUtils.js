import React from "react";
import { ThemeProvider } from "styled-components";
import { mount, shallow } from "enzyme";
import { theme as defaultTheme } from "../lib/dracs.es.js";

export const shallowWithTheme = (children, customTheme) => {
  const theme = customTheme || defaultTheme;
  return shallow(children, { context: theme });
};

/* eslint-disable */
export function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    ),
  });
}

//FOR ACCESSING PORTALS VIA ENZYME -- import portal from dracs
// https://github.com/airbnb/enzyme/issues/1202#issuecomment-351639783
// function mountPortal(el){
//   const wrapper =  mountWithTheme(el);
//   wrapper.setState({ visible: true});
//   const portalInstance = wrapper.find(Portal).instance();
//   return mountWithTheme(portalInstance.props.children);
// }

//FOR ACCESSING EVENT LISTENERS ATTACHED TO DOCUMENT OR DOCUMENT.BODY
//https://github.com/airbnb/enzyme/issues/426#issuecomment-296304015
// const map = {};
// document.body.addEventListener = jest.fn((event, cb) => {
//   map[event] = cb;
// });
