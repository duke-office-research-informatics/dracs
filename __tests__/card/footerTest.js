import React from "react";
import { mountWithTheme } from "../../config/scUtils.js";
import { axe } from "../../config/axe-helper.js";
import { CardFooter, theme } from "../../lib/dracs.es.js";

describe("CardFooter", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mountWithTheme(<CardFooter />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("returns no a11y violations", async () => {
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });

  it("sets bgColor via props", () => {
    wrapper.setProps({ bgColor: theme.colors.muted });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("footer__CardFooterWrap")).toHaveStyleRule(
      "background-color",
      theme.colors.muted
    );
  });

  it("sets a border-top when border is set to true", () => {
    wrapper.setProps({ border: true });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("footer__CardFooterWrap")).toHaveStyleRule(
      "border-top",
      `solid 1px ${theme.colors.border}`
    );
  });

  it("sets proper styles when `multipleActions` set to true", () => {
    wrapper.setProps({ multipleActions: true });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("footer__CardFooterWrap")).toHaveStyleRule(
      "justify-content",
      "space-between"
    );
  });

  it("applies a style object via props", () => {
    wrapper.setProps({ style: { color: theme.colors.action } });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("footer__CardFooterWrap").props().style).toHaveProperty(
      "color",
      theme.colors.action
    );
  });
});
