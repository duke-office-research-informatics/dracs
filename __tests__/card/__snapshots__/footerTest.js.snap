// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFooter applies a style object via props 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  width: 100%;
  padding: calc(4px);
  border-top: solid 1px #e8e5e2;
  background-color: transparent;
  overflow: hidden;
}

<footer__CardFooterWrap
  border={true}
  style={
    Object {
      "color": "#0680cd",
    }
  }
>
  <footer
    className="c0"
    style={
      Object {
        "color": "#0680cd",
      }
    }
  />
</footer__CardFooterWrap>
`;

exports[`CardFooter matches snapshot 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  width: 100%;
  padding: calc(4px);
  border-top: solid 1px #e8e5e2;
  background-color: transparent;
  overflow: hidden;
}

<footer__CardFooterWrap
  border={true}
>
  <footer
    className="c0"
  />
</footer__CardFooterWrap>
`;

exports[`CardFooter sets a border-top when border is set to true 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  width: 100%;
  padding: calc(4px);
  border-top: solid 1px #e8e5e2;
  background-color: transparent;
  overflow: hidden;
}

<footer__CardFooterWrap
  border={true}
>
  <footer
    className="c0"
  />
</footer__CardFooterWrap>
`;

exports[`CardFooter sets bgColor via props 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  width: 100%;
  padding: calc(4px);
  border-top: solid 1px #e8e5e2;
  background-color: #696969;
  overflow: hidden;
}

<footer__CardFooterWrap
  bgColor="#696969"
  border={true}
>
  <footer
    className="c0"
  />
</footer__CardFooterWrap>
`;

exports[`CardFooter sets proper styles when \`multipleActions\` set to true 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  width: 100%;
  padding: calc(4px);
  border-top: solid 1px #e8e5e2;
  background-color: transparent;
  overflow: hidden;
}

<footer__CardFooterWrap
  border={true}
  multipleActions={true}
>
  <footer
    className="c0"
  />
</footer__CardFooterWrap>
`;
